shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -283,8 +286,12 @@ private CharSequence getClipboardData(PlatformChannel.ClipboardContentFormat for
 
     if (!clipboard.hasPrimaryClip()) return null;
 
+    ClipDescription clipDescription = clipboard.getPrimaryClipDescription();
+    if (clipDescription == null || !clipDescription.hasMimeType(MIMETYPE_TEXT_PLAIN)) {
```

Comment:
I suspect what we really want to do here is to use https://developer.android.com/reference/android/content/ClipData.Item#coerceToText(android.content.Context) - wdyt?

Reply:
@dnfield i think it's SecurityException in `clip.getItemAt(0).coerceToText(activity) ` If getUri() is non-null and try to retrieve its data as a text stream from its content provider  :

> W/ClipData(29672): java.lang.SecurityException: Permission Denial: reading com.android.providers.media.MediaProvider uri content://media/external/file/125 from pid=29672, uid=10307 requires android.permission.READ_EXTERNAL_STORAGE, or grantUriPermission()

So that's why i have to check if it's a text : 

```
ClipDescription clipDescription = clipboard.getPrimaryClipDescription();
    if (clipDescription == null || !clipDescription.hasMimeType(MIMETYPE_TEXT_PLAIN)) {
      return null;
    }
```

Reply:
I think checking for the permission is pretty reasonable. I would say that if the app doesn't have permission, it would probably be helpful to have a debug log in there informing the developer that they should add it to their application to support this kind of copying and pasting. 

What if the clip is the intent type?

Reply:
> What if the clip is the intent type?

The intent will be converted to String without any problem 

<details>
  <summary>coerceToText Source Code</summary>

```
  public CharSequence coerceToText(Context context) {
            // If this Item has an explicit textual value, simply return that.
            CharSequence text = getText();
            if (text != null) {
                return text;
            }

            // If this Item has a URI value, try using that.
            Uri uri = getUri();
            if (uri != null) {
                // First see if the URI can be opened as a plain text stream
                // (of any sub-type).  If so, this is the best textual
                // representation for it.
                final ContentResolver resolver = context.getContentResolver();
                AssetFileDescriptor descr = null;
                FileInputStream stream = null;
                InputStreamReader reader = null;
                try {
                    try {
                        // Ask for a stream of the desired type.
                        descr = resolver.openTypedAssetFileDescriptor(uri, "text/*", null);
                    } catch (SecurityException e) {
                        Log.w("ClipData", "Failure opening stream", e);
                    } catch (FileNotFoundException|RuntimeException e) {
                        // Unable to open content URI as text...  not really an
                        // error, just something to ignore.
                    }
                    if (descr != null) {
                        try {
                            stream = descr.createInputStream();
                            reader = new InputStreamReader(stream, "UTF-8");

                            // Got it...  copy the stream into a local string and return it.
                            final StringBuilder builder = new StringBuilder(128);
                            char[] buffer = new char[8192];
                            int len;
                            while ((len=reader.read(buffer)) > 0) {
                                builder.append(buffer, 0, len);
                            }
                            return builder.toString();
                        } catch (IOException e) {
                            // Something bad has happened.
                            Log.w("ClipData", "Failure loading text", e);
                            return e.toString();
                        }
                    }
                } finally {
                    IoUtils.closeQuietly(descr);
                    IoUtils.closeQuietly(stream);
                    IoUtils.closeQuietly(reader);
                }

                // If we couldn't open the URI as a stream, use the URI itself as a textual
                // representation (but not for "content", "android.resource" or "file" schemes).
                final String scheme = uri.getScheme();
                if (SCHEME_CONTENT.equals(scheme)
                        || SCHEME_ANDROID_RESOURCE.equals(scheme)
                        || SCHEME_FILE.equals(scheme)) {
                    return "";
                }
                return uri.toString();
            }

            // Finally, if all we have is an Intent, then we can just turn that
            // into text.  Not the most user-friendly thing, but it's something.
            Intent intent = getIntent();
            if (intent != null) {
                return intent.toUri(Intent.URI_INTENT_SCHEME);
            }

            // Shouldn't get here, but just in case...
            return "";
        }
```

</details>

- URI
If ClipData.Item is a URI (getUri() is not null), coerceToText() tries to use it as a content URI:
If the URI is a content URI and the provider can return a text stream, coerceToText() returns a text stream.
If the URI is a content URI but the provider does not offer a text stream, coerceToText() returns a representation of the URI. The representation is the same as that returned by Uri.toString().
If the URI is not a content URI, coerceToText() returns a representation of the URI. The representation is the same as that returned by Uri.toString().

- Intent
If ClipData.Item is an Intent (getIntent() is not null), coerceToText() converts it to an Intent URI and returns it. The representation is the same as that returned by Intent.toUri(URI_INTENT_SCHEME).

Source : [CoerceToText](https://developer.android.com/guide/topics/text/copy-paste#CoerceToText)

Reply:
@dnfield FYI : 

https://github.com/flutter/flutter/issues/66108#issuecomment-699886651

Thanks